app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,33 @@
+class CommentsController < ApplicationController
+  def new
+    puts("params #{params}")
+    @post = Post.find(params[:post_id])
+    @user = ApplicationController.new.current_user
+    @comment = @user.comments.new
+    render :new, locals: { comment: @comment }
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @user = ApplicationController.new.current_user
+    add_comment = Comment.create(author: @user, post: @post, text: comment_params['text'])
```

Comment:
- [ ] please review the previous note this can be applied to this part of the code as well. the only addition that needs to be made is in the create method the comment is associated with the post by doing ` @comment.post_id = params[:post_id]`